ci: build and deploy benchmark images with jib

oleschoenburg opened the pull request:
- Build benchmark worker and starter images with jib
- Build and push worker and starter images with `latest` tag for every new commit on `main`
- Build and push worker and starter images when creating new benchmarks
- Configure benchmarks to use newly created worker and starter images

relates to #11484

oleschoenburg commented:
The new release script will not work for older versions of course. To do a release of the starter/worker images for versions which do not yet use jib, we'd have to use an older version of the release script.

Zelldon commented on .github/workflows/benchmark.yml:
```diff
@@ -86,10 +86,50 @@ jobs:
+        id: auth
+        with:
+          token_format: 'access_token'
+          workload_identity_provider: 'projects/628707732411/locations/global/workloadIdentityPools/zeebe-gh-actions/providers/gha-provider'
```
:thinking: Wondering whether this is necessary, since we already did this above? Is it because it is a different job?

oleschoenburg replied:
Yep, that's it

Zelldon commented on benchmarks/project/README.md:
````diff
@@ -4,16 +4,11 @@ This project contains code for the starter and worker, which are used during our
-docker-compose push
+```bash
+./mvnw -am -pl benchmarks/project package -DskipTests -DskipChecks
+./mvnw -pl benchmarks/project jib:build -Pstarter
````
:rocket:

Zelldon replied:
Be aware that we might need to change the release process and mention this as well, right? Or is this now automatically done on release? I think you haven't changed that ?

oleschoenburg replied:
Right, let me check where we mention this in the release process

oleschoenburg replied:
The release process only [mentions](https://github.com/zeebe-io/zeebe-engineering-processes/blob/f6d4acb4e4d4b996c6b7b7f06ffbed0caf909c4d/src/main/resources/release/post_release.bpmn#L108) the `release.sh` script here, which I've adjusted.

Zelldon reviewed (approved):
Great thanks @oleschoenburg :rocket:

oleschoenburg commented:
I rebased and then tested again.
I noticed that the way we configure starter and worker does not work anymore: https://github.com/zeebe-io/benchmark-helm/pull/20

I also adjusted the default tag to be "SNAPSHOT" instead of "latest" so that we don't have to change that too in benchmark-helm.

oleschoenburg commented:
bors r+

oleschoenburg commented:
bors merge

npepinpe commented:
so...when are we migrating zeebe itself to jib? :eyes:
